### ISSUE TITLE
Fix the workflow for revoking a block on a user (un-blocking)

### ISSUE THREAD

**Issue by palisadoes** [PalisadoesFoundation/talawa-api#1863]
**Is your feature request related to a problem? Please describe.**

We don't have a clearly defined workflow for blocking users

**Describe the solution you'd like**

1. A blocked user cannot be a member of an `Organization`, but they can be listed in an `Organization`'s `blockedUsers` field
2. Blocked users for an `Organization` must not be able to send a `membershipRequest` to that `Organization`
3. When a user is unblocked:
     1. if registration approval is required a `membershipRequest` should be sent to the `Organization` unblocking the user
     1. if registration approval is not required, then the user should be added to the members field of the `Organization`.

**Describe alternatives you've considered**

- N/A

**Approach to be followed (optional)**

- N/A

**Additional context**
1. This need was identified here:
    1. https://github.com/PalisadoesFoundation/talawa-api/pull/1770
1. Screen shot of the `Organization` collection

    ![image](https://github.com/PalisadoesFoundation/talawa-api/assets/16875803/f2d34a01-3b2e-4709-8e02-1649d0cbed0d)

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

**Comment by Sahi1l-Kumar** [comment]
Hello. I would like to work on this issue.

**Comment by Sahi1l-Kumar** [comment]
@palisadoes Can I change the schema for `sendMembershipRequest ` mutation to accept `userId` as a field so that in the `sendMembershipRequest resolver` I can do the `check` for `If the user exists` and solve this part effectively

>2.Blocked users for an `Organization` must not be able to send a `membershipRequest` to that `Organization`

**Comment by Sahi1l-Kumar** [comment]
There were a few cases which were not covered in the solution. For example:-
If an `organization` has `userRegistrationRequired` to be `false` even then a`blocked user` is still able to join the `organization` using the `JoinPublicOrganization` mutation. 
I am not sure if this is out of scope of this issue. Should I cover these cases as well and also do i need to add tests for the above pr?

**Comment by palisadoes** [comment]
That's not a desired blocked user outcome and should be a part of the fix.

**Comment by Sahi1l-Kumar** [comment]
Ok. I will try to add all such cases to the fix
